mllib/src/main/scala/org/apache/spark/ml/clustering/GaussianMixture.scala

```diff
@@ -105,6 +108,15 @@ class GaussianMixtureModel private[ml] (
   def gaussians: Array[MultivariateGaussian] = parentModel.gaussians
 
   @Since("2.0.0")
```

Comment:
I agree it simplifies things when the return result is either a basic type or a DataFrame.  I think it makes sense for topics and synonyms, for which there is no "natural" representation.  But for distributions, there is a natural class to provide (the MultivariateGaussian class, with its associated methods like pdf).  I'd also be OK with providing only a dataframe for now in Python, perhaps using a method called ```gaussiansDF```.